### PR TITLE
Force a Ninja regeneration before each project test

### PR DESCRIPTION
**Pull request by nirbheek**
We have no test coverage for regeneration at all, which is why issues like #1246 slide by without us noticing. With this, we will run a regen on every test during `ninja test` after it has been compiled. This will not affect test times too much since the regen will not rebuild anything at all since there have been no source changes.

### CODE REVIEW

File: mesonbuild/dependencies.py [hunk 1]
```diff
@@ -1152,21 +1193,27 @@ def __init__(self, environment, kwargs):
...
                 self.cargs = pcdep.get_compile_args()
                 self.linkargs = pcdep.get_link_args()
+                self.version = pcdep.get_version()
                 return
         except Exception:
             pass
         if mesonlib.is_osx():
             self.is_found = True
             self.linkargs = ['-framework', 'OpenGL']
+            self.version = '1' # FIXME
```
Comment (ignatenkobrain) [inline mesonbuild/dependencies.py]: __[flake8]__
* [E261] at least two spaces before inline comment

File: mesonbuild/dependencies.py [hunk 2]
```diff
@@ -1152,21 +1193,27 @@ def __init__(self, environment, kwargs):
...
             return
         if mesonlib.is_windows():
             self.is_found = True
             self.linkargs = ['-lopengl32']
+            self.version = '1' # FIXME: unfixable?
```
Comment (ignatenkobrain) [inline mesonbuild/dependencies.py]: __[flake8]__
* [E261] at least two spaces before inline comment

File: mesonbuild/dependencies.py (hunk 1 above)
Comment (nirbheek) [inline mesonbuild/dependencies.py]: Turn that one off, honestly. Such a ridiculous one.

**Comment by nirbheek** [comment]
Yay, we've already found one bug. Also fixed in the next commit:

```
Traceback (most recent call last):
  File "/root/mesonbuild/mesonmain.py", line 292, in run
    app.generate()
  File "/root/mesonbuild/mesonmain.py", line 177, in generate
    intr.run()
  File "/root/mesonbuild/interpreter.py", line 2222, in run
    super().run()
  File "/root/mesonbuild/interpreterbase.py", line 124, in run
    self.evaluate_codeblock(self.ast, start=1)
  File "/root/mesonbuild/interpreterbase.py", line 145, in evaluate_codeblock
    raise e
  File "/root/mesonbuild/interpreterbase.py", line 139, in evaluate_codeblock
    self.evaluate_statement(cur)
  File "/root/mesonbuild/interpreterbase.py", line 152, in evaluate_statement
    return self.assignment(cur)
  File "/root/mesonbuild/interpreterbase.py", line 547, in assignment
    value = self.evaluate_statement(node.value)
  File "/root/mesonbuild/interpreterbase.py", line 150, in evaluate_statement
    return self.function_call(cur)
  File "/root/mesonbuild/interpreterbase.py", line 371, in function_call
    return self.funcs[func_name](node, self.flatten(posargs), kwargs)
  File "/root/mesonbuild/interpreter.py", line 1703, in func_dependency
    found = cached_dep.get_version()
AttributeError: 'WxDependency' object has no attribute 'get_version'
ninja: error: rebuilding 'build.ninja': subcommand failed
```

File: mesonbuild/dependencies.py (hunk 1 above)
Comment (ignatenkobrain) [inline mesonbuild/dependencies.py]: yep, agreed!

**Pull request merged**